apache/superset
# feat: Flow for tables that already have a dataset

### SUMMARY

This implements a flow in the new dataset creation page for when a table already has a dataset. It uses a simplified dataset fetch in the `AddDataset/index.tsx` file to cross-reference with the table list. If the table has a dataset, it gets a warning icon in the left panel. If the table is selected, the table's columns are still displayed, but the "Create Dataset" button is disabled and it has an info banner at the top informing the user of the pre-existing dataset with a link to that dataset in the upper-left corner of the alert. Clicking the "View Dataset" button in this alert will bring the user to the explore view of the existing dataset in a new tab.

### ANIMATED GIF / SCREENSHOT

#### Screenshot of selected table with existing dataset
<img width="1388" alt="Screenshot 2022-11-15 at 7 01 47 PM" src="https://user-images.githubusercontent.com/55605634/202057625-a50027ee-bd71-483f-a32e-b3cebfa83990.png">

#### Left panel warning icons
![existingDSleftPanel](https://user-images.githubusercontent.com/55605634/202057245-1debb1d0-d6db-4b3a-8390-1b23ccd70102.gif)

#### "View dataset"
![existingDSViewDS](https://user-images.githubusercontent.com/55605634/202057751-40b064d7-e2fb-4f86-9502-f3dcbcaa7327.gif)


### TESTING INSTRUCTIONS

- Go to `http://localhost:9000/dataset/add/?testing`
- Select a database and a schema
- Observe that any tables in the left panel with a pre-existing dataset will have a warning icon
- Click a table with a dataset
- Observe that the "Create dataset" button is disabled and there is an info banner at the top informing of the pre-existing dataset
- Click "View dataset" in the alert
- Observe that you are taken to the explore view of the existing dataset in a new tab

### ADDITIONAL INFORMATION
